storage/store.go

```diff
@@ -900,7 +900,7 @@ func (s *Store) RemoveRange(rng *Range) error {
 	n := sort.Search(len(s.rangesByKey), func(i int) bool {
 		return bytes.Compare(rng.Desc().StartKey, s.rangesByKey[i].Desc().EndKey) < 0
 	})
-	if n >= len(s.rangesByKey) {
+	if n >= len(s.rangesByKey) || rng.Desc().RaftID != s.rangesByKey[n].Desc().RaftID {
```

Comment:
I wonder if we should be using an ordered tree (like `RangeCache`) instead of a sorted slice of ranges. I'll add it to my TODO list to investigate.